### PR TITLE
rubysrc2cpg: Refactor if statement

### PR DESCRIPTION
**Pull request by xavierpinho**
* Reworked the previous `else` handling (that was being placed under a `JUMP_TARGET` node), to be placed under a `BLOCK` node.
* Removed an incorrect unit-test
* Ignored a unit-test for `when-case-else` expressions, as they also need to be refactored and improved.

### CODE REVIEW

File: joern-cli/frontends/rubysrc2cpg/src/main/scala/io/joern/rubysrc2cpg/astcreation/AstCreator.scala [hunk 1]
```diff
@@ -497,8 +497,10 @@ class AstCreator(
       })
       .toList
 
-    val stmtAsts = whenThenAstsList ++ Option(ctx.caseExpression().elseClause()).map(astForElseClause).getOrElse(Seq())
-    val block    = blockNode(ctx.caseExpression())
+    val stmtAsts = whenThenAstsList ++ Option(ctx.caseExpression().elseClause())
+      .map(ctx => astForCompoundStatement(ctx.compoundStatement()))
```
Comment (ursachec) [inline joern-cli/frontends/rubysrc2cpg/src/main/scala/io/joern/rubysrc2cpg/astcreation/AstCreator.scala:501]: just a nit
```suggestion
     .map(astForCompoundStatement(_.compoundStatement()))
```

File: joern-cli/frontends/rubysrc2cpg/src/main/scala/io/joern/rubysrc2cpg/astcreation/AstForExpressionsCreator.scala [hunk 2]
```diff
@@ -153,31 +153,20 @@ trait AstForExpressionsCreator { this: AstCreator =>
   protected def astForIfExpression(ctx: IfExpressionContext): Ast = {
     val testAst   = astForExpressionOrCommand(ctx.expressionOrCommand())
     val thenAst   = astForCompoundStatement(ctx.thenClause().compoundStatement())
-    val elsifAsts = Option(ctx.elsifClause()).map(_.asScala).getOrElse(Seq()).map(astForElsifClause)
-    val elseAst   = Option(ctx.elseClause()).map(astForElseClause).getOrElse(Seq())
-    val ifNode    = controlStructureNode(ctx, ControlStructureTypes.IF, ctx.getText)
+    val elsifAsts = Option(ctx.elsifClause).map(_.asScala).getOrElse(Seq()).map(astForElsifClause)
+    val elseAst = Option(ctx.elseClause()).map(ctx => astForCompoundStatement(ctx.compoundStatement())).getOrElse(Seq())
```
Comment (ursachec) [inline joern-cli/frontends/rubysrc2cpg/src/main/scala/io/joern/rubysrc2cpg/astcreation/AstForExpressionsCreator.scala:157]: again, just a nit
```suggestion
    val elseAst = Option(ctx.elseClause()).map(astForCompoundStatement(_.compoundStatement())).getOrElse(Seq())
```

File: joern-cli/frontends/rubysrc2cpg/src/main/scala/io/joern/rubysrc2cpg/astcreation/AstForStatementsCreator.scala [hunk 3]
```diff
@@ -99,11 +99,15 @@ trait AstForStatementsCreator { this: AstCreator =>
     controlStructureAst(throwNode, rhs.headOption, lhs)
   }
 
-  protected def astForCompoundStatement(ctx: CompoundStatementContext): Seq[Ast] =
-    Option(ctx.statements()).map(astForStatements).getOrElse(Seq())
+  protected def astForCompoundStatement(ctx: CompoundStatementContext): Seq[Ast] = {
+    val stmtAsts   = Option(ctx.statements()).map(astForStatements).getOrElse(Seq())
+    val blockNode_ = blockNode(ctx)
+    Seq(Ast(blockNode_).withChildren(stmtAsts))
```
Comment (ursachec) [inline joern-cli/frontends/rubysrc2cpg/src/main/scala/io/joern/rubysrc2cpg/astcreation/AstForStatementsCreator.scala:105]: there's a `blockAst` fn you can use

**Review by ursachec — APPROVED** [review]

File: joern-cli/frontends/rubysrc2cpg/src/main/scala/io/joern/rubysrc2cpg/astcreation/AstCreator.scala (hunk 1 above)
Comment (xavierpinho) [reply joern-cli/frontends/rubysrc2cpg/src/main/scala/io/joern/rubysrc2cpg/astcreation/AstCreator.scala:501]: I like nits, but it's not the same expansion: your suggestion would expand to (modulo alpha-conversion..)

```scala
.map(astForCompoundStatement(x => x.compoundStatement()))
```

Or am I missing something?

Comment (ursachec) [reply joern-cli/frontends/rubysrc2cpg/src/main/scala/io/joern/rubysrc2cpg/astcreation/AstCreator.scala:501]: ohh, you're right, just saw it. ok nvm

**Pull request merged**